### PR TITLE
Lay foundations

### PR DESCRIPTION
**Pull request by wuletawwonte**
Changes in this PR
- Initialize the project using `create-react-app`
- Add three pages including the `NoMatch` page
- Add routing for all the pages
- Organize the files in `components` and `pages` folders 
- Fix all linter errors
- Use functional components instead of class based components

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,8 +6,10 @@
     "@testing-library/jest-dom": "^5.16.4",
     "@testing-library/react": "^13.1.1",
     "@testing-library/user-event": "^13.5.0",
+    "prop-types": "^15.8.1",
     "react": "^18.1.0",
     "react-dom": "^18.1.0",
+    "react-router-dom": "^6.3.0",
```
Comment (Cathella) [inline package.json:12]: ✅ `react-router-dom` should be present in `package.json` 👍🏻.

File: src/App.js [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
-  );
-}
+const App = () => (
+  <div className="main-container">
+    <Navbar />
+    <Routes>
+      <Route path="/" exact element={<BooksPage />} />
+      <Route path="/categories" element={<CategoriesPage />} />
+      <Route path="*" element={<NoMatch />} />
+    </Routes>
```
Comment (Cathella) [inline src/App.js:16]: ✅ Has 2 `<Route>`s, one for the BookList/BooksPage component and the other for the CategoriesPage component 👍🏻.

File: src/components/BooksList.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        {books.map((book) => (
+          <Book
+            key={book.id}
+            id={book.id}
+            name={book.name}
+            author={book.author}
+          />
+        ))}
+      </ul>
+    </div>
```
Comment (Cathella) [inline src/components/BooksList.js:37]: ✅ BookList renders a list of books using the Book component 👍🏻.

File: src/components/Book.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+const Book = (props) => {
+  const { id, name, author } = props;
+  return (
+    <li key={id}>
+      <p>{name}</p>
+      <p>{author}</p>
+      <button type="button">Remove</button>
+    </li>
+  );
+};
```
Comment (Cathella) [inline src/components/Book.js:13]: ✅ The Book component takes the book data as a prop and renders the title and the author 👍🏻 
✅ Has a button for removing books 👍🏻

File: src/components/Navbar.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+const Navbar = () => (
+  <div className="navbar">
+    <h1>Bookstore CMS</h1>
+    <nav>
+      <ul className="menu-links">
+        {links.map((link) => (
+          <li key={link.id}>
+            <Link to={link.path}>{link.text}</Link>
+          </li>
+        ))}
```
Comment (Cathella) [inline src/components/Navbar.js:14]: ✅ Has 2 `<Link>`s in the navbar which leads to the booklist and categories locations 👍🏻.

File: src/components/RegisterBook.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+const RegisterBook = () => (
+  <>
+    <h2>Register Book</h2>
+    <form className="register-book-form">
+      <input type="text" name="title" placeholder="Title" />
+      <input type="text" name="author" placeholder="Author" />
+      <button type="submit">Register</button>
+    </form>
+  </>
+);
```
Comment (Cathella) [inline src/components/RegisterBook.js:12]: ✅ RegisterBook component renders a form with inputs for the title and the author of the book 👍🏻 
✅ Has a button to submit 👍🏻

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @wuletawwonte 👋🏻, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💪🏻

✅ No linter errors.
✅ You used correct GitHub Flow.
✅ Professional README.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by wuletawwonte** [comment]
Thank you @Cathella For the good review.

**Pull request merged**